### ISSUE TITLE
How to prepare data for training ?

### ISSUE THREAD

**Issue by nehaleosharma** [emredogan7/age-estimation-by-CNN#1]
Hello, 
I was going through your code to convert the dataset into .npy file format and came across the code lines 
path_train = "./../data/train/"
path_test = "./../data/test/"
path_validation = "./../data/validation/"

Do I have to split the UTKFace dataset into Train Test and Validation sets by my self and put those into respective folders before running the read_data.py?

Also, I would like to know that where was your paper published as I want to use it as a reference. I would be glad if you could share your publication details. 
Thanks in Advance. Pls Reply ASAP.

**Comment by mrdogan7** [comment]
Hi there,
First of all, thanks for your interest.

This study belongs to the assignment of our Deep Learning course, not a published paper.  So, it is not possible for me to share the publication details. But if you are interested, you can check the technical report [here](https://github.com/mrdogan7/age-estimation-by-CNN/blob/master/doc/report.pdf).

About your first question, we used a downsampled version of UTKFace dataset shared by the instructor of the course. For our assignment, we were supposed to split data into training, validation and test sets. It is not strongly necessary but it would be better to use a validation set in order to optimize hyperparameters. 
To speed up the data reading process, we converted image files to .npy data. We cannot share them here for the storage limitations but, if you want to access these .npy files, you can check [this link](https://drive.google.com/open?id=1GD9K7WSZJL4t9UAXi_ZhIsWhPsoOTGEk).

In [read_data.py](https://github.com/mrdogan7/age-estimation-by-CNN/blob/master/read_data.py), the relative paths for data sets are given wrong. Instead they are supposed to be 

`path_train = "./data/train/"`
`path_test = "./data/test/"`
`path_validation = "./data/validation/"`
asuming that your `data` folder is created in the absolute path of the repo.

And yes, if you want to try the code for a new dataset, you are supposed to create  `data` folder and `train`, `validation`, `test` subfolders within it before you run `read_data.py`. 

Please don't hesitate to ask any further questions.

**Comment by nehaleosharma** [comment]
hello
 Thank you so much for your immediate reply .I am working on this project.
I will surely ask further queries.

Thanks
